include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -300,6 +301,19 @@ class IExecutionProvider {
    */
   virtual bool ConcurrentRunSupported() const { return true; }
 
+  /** Some session option values (default or user provided) may not work with some EPs.
+   * The EP should check and assure the SessionOptions object fit its implementation to avoid potential crash.
+   * we would rather throw errors to user and remind of setting appropriate config.
+   *
+   * @param session_options The SessionOptions object to be altered.
+   * @return Status indicating whether the SessionOptions object is valid for the EP or not.
+   * The EP should return an error if it's incompatible with the SessionOptions that specifies
+   * which option is problematic so the user knows what needs to be fixed
+   */
+  virtual common::Status CheckSessionOptionsAndProcess(const SessionOptions& /*session_options*/) {
```

Comment:
How does someone writing a new EP know which options they should care about to implement this?

Reply:
I think it's EP author's onus to figure out which options is not compatiable to this new EP. After all, we can't rely on the framework or ORT users to take this responsibility.
A new EP should handle all conflicts  with sessionOption, to either acommdate EP-self or report a error.

May we could document all sessionOptions in the how-to-guide, `add a new EP`.